python/ray/_private/thirdparty/pyamdsmi/pyamdsmi.py

```diff
@@ -0,0 +1,552 @@
+# MIT License
```

Comment:
Is this library also available on pypi? How should we upgrade it in the future?

Reply:
It is not available on pypi. This proposed implementation in pyamdsmi.py will require manual code updates if there are changes in the future. But we do have an alternate implementation that provides offline installation through pip if ROCm is already installed on the system. See https://github.com/ROCm/amdsmi/tree/develop#manualmultiple-rocm-instance-python-library-install. Please let us know which implementation you would prefer or if further discussion is required.